R/fitNullModel.R

```diff
@@ -193,3 +221,83 @@ nullModelSmall <- function(null.model) {
 isNullModelSmall <- function(null.model) {
     is.null(null.model$cholSigmaInv)
 }
+
+
+.modelString <- function(outcome, covars = NULL, random = NULL, group.var = NULL, inverse_normal = FALSE) {
+    model.outcome <- .modelOutcomeString(outcome, inverse_normal = inverse_normal)
+    model.covars <- if (is.null(covars)) NULL else .modelCovarString(covars)
+    model.random <- if (is.null(random)) NULL else paste(paste0("(1|", random, ")"), collapse=" + ")
+    model.var <- if (is.null(group.var)) NULL else paste0("var(", group.var, ")")
+    model.string <- paste(c(model.covars, model.random, model.var), collapse=" + ")
+    paste(model.outcome, model.string, sep=" ~ ")
+}
+
+.modelCovarString <- function(covars) {
+  paste(covars, collapse=" + ")
+}
+
+.modelOutcomeString <- function(outcome, inverse_normal = FALSE) {
+  if (inverse_normal) {
+    return(sprintf("rankInvNorm(resid(%s))", outcome))
+  } else {
+    return(outcome)
+  }
+}
+
+.setCovMatNames <- function(cov.mat) {
+  if (!is.list(cov.mat)) {
+    cov.mat <- list(cov.mat)
+  }
```

Comment:
I think that `.covMatNames` turns a list of length 1 into a matrix, then this function turns it back into a list. That might not be a problem, but I wonder if we can/should clean up all this covMat manipulation code.

Reply:
It looks like `.covMatNames`  doesn't return the matrix; it returns the sample.ids associated with the matrix. It also does some name checking to make sure that the rownames and colnames (for a given matrix in the list?) are the same. So I think it's ok for this update but we could look into whether any of the covMat code can be cleaned up in a new branch.

Reply:
sounds good